File: Day124_2025/Q124_GFG.java
class Q124_GFG {
    public int findSubString(String str) {
        int n = str.length();
        int uniqueCount = (int) str.chars().distinct().count();

        int[] freq = new int[256];
        int left = 0, right = 0, ans = n;
        int count = 0;

        while (right < n) {
            if (freq[str.charAt(right)] == 0) count++;
            freq[str.charAt(right)]++;
            right++;

            while (count == uniqueCount) {
                ans = Math.min(ans, right - left);
                freq[str.charAt(left)]--;
                if (freq[str.charAt(left)] == 0) count--;
                left++;
            }
        }

        return ans;
    }
}

File: Day124_2025/Q124_GFG.py
class Solution:
    def findSubString(self, s):
        unique_chars = set(s)
        required = len(unique_chars)
        freq = {}
        left = 0
        ans = len(s)

        for right in range(len(s)):
            freq[s[right]] = freq.get(s[right], 0) + 1

            while len(freq) == required:
                ans = min(ans, right - left + 1)
                freq[s[left]] -= 1
                if freq[s[left]] == 0:
                    del freq[s[left]]
                left += 1

        return ans
